### ISSUE TITLE
MotS - OpenJKDF2 loading different textures when material files are present

### ISSUE THREAD

**Issue by Vortikai** [shinyquagsire23/OpenJKDF2#233]
When investigating #230 using JKGFXMod without OpenMOTS, we discovered that for whatever reason OpenMOTS loads a different texture for the elevator call panel button. When mat files are present in the materials folder under \<motsdirecotry>\jkgm\materials, such as jknup and motsnup (jkgfxmods that have upscaled the textures using AI), the texture 00t_4.mat is loaded from the original JK game instead of 00t_4.mat from MotS

![openmotssecurirtycamerapanel1](https://media.discordapp.net/attachments/800417541654577204/1074768868851064973/image.png?width=750&height=567)
_OpenMOTS loads this texture, non-default/vanilla, when jknup is present_

![openmotssecurirtycamerapanel2](https://media.discordapp.net/attachments/800417541654577204/1074768921640587324/image.png?width=750&height=567)
_Default MOTS/jkgfxmod disabled in OpenMOTS, vanilla, also when jknup is not present_

This affects other levels as well, such as the control panels for security camera stations on level 3 - Asteroid Core

![openmotssecurirtycamerapanel3](https://cdn.discordapp.com/attachments/800417541654577204/1074793446067404800/20230213224433_1.jpg)
_This is the default look of the game without jkgfxmod enabled in OpenMOTS_

![openmotssecurirtycamerapanel4](https://cdn.discordapp.com/attachments/800417541654577204/1074772499662917662/20230213212158_1.jpg)
_When jkgfxmod is enabled with jknup and motsnup in OpenMOTS_

In MotS (not OpenMOTS), jkgfxmod seems to be loading 00t_8.mat in the first level where it's got 00t_4.mat as the elevator call button, because it's also loading the emissive textures for 00t_8.mat while at it
This is probably because since 00t_4.mat and 00t_8.mat are identical in MotS, jkgfxmod loads 00t_8.mat's texture signatures before checking what 00t_4.mat actually is
(Texture signatures from metadata.json, like this
            "replaces_signatures": [
                "26b4cb5304ec1eb136ed57953dfb5f58",
                "471b46d2301fd856904a6c9bff0db878",
                "cedefb83eaf622b944dd54824a39ebf3",
                "ff2f7eb776d308a0ecb2882194d52dbe"
            ])

We don't really now though.

In short, OpenMOTS is loading some textures, for now the only confirmed one being elevator call panels/security station buttons, from the old JK side of things and not the MOTS side of things. It isn't an issue with a vanilla install or a jkgfxmod build.

Toggling jkgfxmod in the options demonstrates the differences clearly while in OpenMOTS.

**Comment by shinyquagsire23** [comment]
Does this happen when the game is launched with `-motsCompat` directly? I'd guess it's just some dirty state sticking around between soft resets.

**Comment by Vortikai** [comment]
Tbh I never figured out -motsCompat launch option stuff and haven't messed with it much. I launch both games separately through Steam. I'll try to figure that out and mess around later. Explicit instructions always appreciated as well

**Comment by Vortikai** [comment]
In extremely simple terms:

![kuva](https://user-images.githubusercontent.com/63983128/218631650-6f9ef9f5-5eae-4ce7-b0e9-0ec237aa7eda.png)

Here's my directory for MotS.
In JKGFXMod only installs with the jknup and motsnup direcotires present, the correct texture loads
![openmotssecurirtycamerapanel2](https://media.discordapp.net/attachments/800417541654577204/1074768921640587324/image.png?width=750&height=567).

With the same directory in OpenMOTS, this incorrect texture loads
![openmotssecurirtycamerapanel1](https://media.discordapp.net/attachments/800417541654577204/1074768868851064973/image.png?width=750&height=567)
When removing jknup, the neural upscaled textures from the original JKDF2 game that MOTS uses, from an OpenMOTS installation, the correct texture displays.

We have no idea why OpenMOTS is (possibly) loading Res2.gob 00t_4.mat instead of JKmres.goo 00t_4.mat

**Comment by shinyquagsire23** [comment]
oh, I actually did all of the MAT stuff based on filenames instead of hashes, since it was faster and I had access to the texture filenames. I guess I'd have to sort out which texture pack wins out in my case.
